internals-js/src/specs/connectSpec.ts

```diff
@@ -128,79 +181,149 @@ export class ConnectSpecDefinition extends FeatureDefinition {
-      URLPathTemplate;
-    ConnectHTTP.addField(new InputFieldDefinition('body')).type = JSONSelection;
-    ConnectHTTP.addField(new InputFieldDefinition('headers')).type =
-      new ListType(new NonNullType(HTTPHeaderMapping));
-
-    ConnectHTTP.addField(new InputFieldDefinition('path')).type = JSONSelection;
-    ConnectHTTP.addField(new InputFieldDefinition('queryParams')).type =
-      JSONSelection;
-
-    connect.addArgument('http', new NonNullType(ConnectHTTP));
-
-    const ConnectBatch  = schema.addType(new InputObjectType(this.typeNameInSchema(schema, CONNECT_BATCH)!));
-    ConnectBatch.addField(new InputFieldDefinition('maxSize')).type = schema.intType();
-    connect.addArgument('batch', ConnectBatch);
-
-    const ConnectorErrors  = schema.addType(new InputObjectType(this.typeNameInSchema(schema, CONNECTOR_ERRORS)!));
-    ConnectorErrors.addField(new InputFieldDefinition('message')).type = JSONSelection;
-    ConnectorErrors.addField(new InputFieldDefinition('extensions')).type = JSONSelection;
-    connect.addArgument('errors', ConnectorErrors);
-
-    connect.addArgument('selection', new NonNullType(JSONSelection));
-    connect.addArgument('entity', schema.booleanType(), false);
 
     /*
-      directive @source(
-        name: String!
-        http: ConnectHTTP
+      directive @connect(
+        source: String
+        http: ConnectHTTP!
+        batch: ConnectBatch
+        selection: JSONSelection!
+        entity: Boolean = false
         errors: ConnectorErrors
-      ) repeatable on SCHEMA
+      ) repeatable on FIELD_DEFINITION
+        | OBJECT # added in v0.2, validation enforced in rust
     */
-    const source = this.addDirective(schema, SOURCE).addLocations(
-      DirectiveLocation.SCHEMA,
+    this.registerDirective(
+      createDirectiveSpecification({
+        name: CONNECT,
+        locations: [DirectiveLocation.FIELD_DEFINITION, DirectiveLocation.OBJECT],
+        repeatable: true,
+        args: [
+          {
+            name: 'source',
+            type: (schema) => schema.stringType()
+          },
+          {
+            name: 'http',
+            type: (schema, feature) => {
+              const connectHttpType =
+                  lookupFeatureTypeInSchema<InputObjectType>(CONNECT_HTTP, 'InputObjectType', schema, feature);
+              return new NonNullType(connectHttpType);
+            }
+          },
+          {
+            name: 'batch',
+            type: (schema, feature) =>
+                lookupFeatureTypeInSchema<InputObjectType>(CONNECT_BATCH, 'InputObjectType', schema, feature)
+          },
+          {
+            name: 'selection',
+            type: (schema, feature) =>
+                lookupFeatureTypeInSchema<ScalarType>(JSON_SELECTION, 'ScalarType', schema, feature)
+          },
+          {
+            name: 'entity',
+            type: (schema) => schema.booleanType(),
+            defaultValue: false
+          },
+          {
+            name: 'errors',
```

Comment:
nit: The `errors` argument is ordered between `batch` and `selection` in the original code.